### ISSUE TITLE
[BUG] https://127.0.0.1:9345/v1-rke2/readyz: 500 Internal Server Error

### ISSUE THREAD

**Issue by pamanseau** [rancher/rancher#45035]
**Rancher Server Setup**
- Rancher version: 2.8.3

**Information about the Cluster**
- OS: Ubuntu 22.04
- Kubernetes version: 1.27.12 & 1.28.8
- Cluster Type (Local/Downstream): CUSTOM

**User Information**
- What is the role of the user logged in? ADMIN/CUSTER

**Describe the bug**

The RANCHER registration script is giving the error 

If uninstall with `/usr/local/bin/rke2-uninstall.sh` and install from `curl -sfL https://get.rke2.io | sh -` RKE2 installation is working fine.

**To Reproduce**

```sh
echo "fs.file-max=1048576" | sudo tee -a /etc/sysctl.conf
sudo sysctl -p
sudo systemctl disable --now ufw
sudo systemctl disable --now apparmor.service
sudo systemctl disable --now unattended-upgrades
```

```sh
curl -fL https://xxx.xxx.com/system-agent-install.sh | sudo  sh -s - --server https://xxx.xxx.com --label 'cattle.io/os=linux' --token 929smr7cfjrcmkvvrrnnljzjqn64tgw22ncqt2d9n54hb755mjcz7p --etcd --controlplane --worker
```

**Result**
```
Apr 07 04:52:40 s12142045 rke2[61201]: time="2024-04-07T04:52:40Z" level=info msg="Using private registry config file at /etc/rancher/rke2/registries.yaml"
Apr 07 04:52:40 s12142045 rke2[61201]: time="2024-04-07T04:52:40Z" level=info msg="Pulling runtime image index.docker.io/rancher/rke2-runtime:v1.27.12-rke2r1"
Apr 07 04:52:41 s12142045 rke2[61201]: time="2024-04-07T04:52:41Z" level=info msg="Creating directory /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/bin"
Apr 07 04:52:41 s12142045 rke2[61201]: time="2024-04-07T04:52:41Z" level=info msg="Extracting file bin/containerd to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/bin/containerd"
Apr 07 04:52:42 s12142045 rke2[61201]: time="2024-04-07T04:52:42Z" level=info msg="Extracting file bin/containerd-shim to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/bin/containerd-shim"
Apr 07 04:52:43 s12142045 rke2[61201]: time="2024-04-07T04:52:43Z" level=info msg="Extracting file bin/containerd-shim-runc-v1 to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/bin/containerd-shim-runc-v1"
Apr 07 04:52:43 s12142045 rke2[61201]: time="2024-04-07T04:52:43Z" level=info msg="Extracting file bin/containerd-shim-runc-v2 to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/bin/containerd-shim-runc-v2"
Apr 07 04:52:43 s12142045 rke2[61201]: time="2024-04-07T04:52:43Z" level=info msg="Extracting file bin/crictl to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/bin/crictl"
Apr 07 04:52:44 s12142045 rke2[61201]: time="2024-04-07T04:52:44Z" level=info msg="Extracting file bin/ctr to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/bin/ctr"
Apr 07 04:52:44 s12142045 rke2[61201]: time="2024-04-07T04:52:44Z" level=info msg="Extracting file bin/kubectl to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/bin/kubectl"
Apr 07 04:52:45 s12142045 rke2[61201]: time="2024-04-07T04:52:45Z" level=info msg="Extracting file bin/kubelet to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/bin/kubelet"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file bin/runc to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/bin/runc"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Creating directory /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/harvester-cloud-provider.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/harvester-cloud-provider.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/harvester-csi-driver.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/harvester-csi-driver.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rancher-vsphere-cpi.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rancher-vsphere-cpi.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rancher-vsphere-csi.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rancher-vsphere-csi.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rke2-calico-crd.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rke2-calico-crd.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rke2-calico.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rke2-calico.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rke2-canal.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rke2-canal.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rke2-cilium.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rke2-cilium.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rke2-coredns.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rke2-coredns.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rke2-flannel.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rke2-flannel.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rke2-ingress-nginx.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rke2-ingress-nginx.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rke2-metrics-server.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rke2-metrics-server.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rke2-multus.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rke2-multus.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rke2-snapshot-controller-crd.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rke2-snapshot-controller-crd.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rke2-snapshot-controller.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rke2-snapshot-controller.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Extracting file charts/rke2-snapshot-validation-webhook.yaml to /var/lib/rancher/rke2/data/v1.27.12-rke2r1-3e47c5e13be8/charts/rke2-snapshot-validation-webhook.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rke2-flannel.yaml to set cluster configuration values"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rke2-snapshot-controller.yaml to set cluster configuration values"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/harvester-cloud-provider.yaml to set cluster configuration values"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/harvester-csi-driver.yaml to set cluster configuration values"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="No cluster configuration value changes necessary for manifest /var/lib/rancher/rke2/server/manifests/rancher/rke2-etcd-snapshot-extra-metadata.yaml"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rancher-vsphere-csi.yaml to set cluster configuration values"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rke2-snapshot-controller-crd.yaml to set cluster configuration values"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rke2-calico-crd.yaml to set cluster configuration values"
Apr 07 04:52:47 s12142045 rke2[61201]: time="2024-04-07T04:52:47Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rke2-canal.yaml to set cluster configuration values"
Apr 07 04:52:48 s12142045 rke2[61201]: time="2024-04-07T04:52:48Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rke2-cilium.yaml to set cluster configuration values"
Apr 07 04:52:48 s12142045 rke2[61201]: time="2024-04-07T04:52:48Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rke2-ingress-nginx.yaml to set cluster configuration values"
Apr 07 04:52:48 s12142045 rke2[61201]: time="2024-04-07T04:52:48Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rke2-metrics-server.yaml to set cluster configuration values"
Apr 07 04:52:48 s12142045 rke2[61201]: time="2024-04-07T04:52:48Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rke2-multus.yaml to set cluster configuration values"
Apr 07 04:52:48 s12142045 rke2[61201]: time="2024-04-07T04:52:48Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rke2-snapshot-validation-webhook.yaml to set cluster configuration values"
Apr 07 04:52:48 s12142045 rke2[61201]: time="2024-04-07T04:52:48Z" level=info msg="No cluster configuration value changes necessary for manifest /var/lib/rancher/rke2/server/manifests/rancher/addons.yaml"
Apr 07 04:52:48 s12142045 rke2[61201]: time="2024-04-07T04:52:48Z" level=info msg="No cluster configuration value changes necessary for manifest /var/lib/rancher/rke2/server/manifests/rancher/cluster-agent.yaml"
Apr 07 04:52:48 s12142045 rke2[61201]: time="2024-04-07T04:52:48Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rancher-vsphere-cpi.yaml to set cluster configuration values"
Apr 07 04:52:48 s12142045 rke2[61201]: time="2024-04-07T04:52:48Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rke2-coredns.yaml to set cluster configuration values"
Apr 07 04:52:48 s12142045 rke2[61201]: time="2024-04-07T04:52:48Z" level=info msg="No cluster configuration value changes necessary for manifest /var/lib/rancher/rke2/server/manifests/rancher/managed-chart-config.yaml"
Apr 07 04:52:48 s12142045 rke2[61201]: time="2024-04-07T04:52:48Z" level=info msg="Updated manifest /var/lib/rancher/rke2/server/manifests/rke2-calico.yaml to set cluster configuration values"
Apr 07 04:52:48 s12142045 rke2[61201]: time="2024-04-07T04:52:48Z" level=info msg="Logging containerd to /var/lib/rancher/rke2/agent/containerd/containerd.log"
Apr 07 04:52:48 s12142045 rke2[61201]: time="2024-04-07T04:52:48Z" level=info msg="Running containerd -c /var/lib/rancher/rke2/agent/etc/containerd/config.toml -a /run/k3s/containerd/containerd.sock --state /run/k3s/containerd --root /var/lib/rancher/rke2/agent/containerd"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="containerd is now running"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Pulling images from /var/lib/rancher/rke2/agent/images/cloud-controller-manager-image.txt"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Image index.docker.io/rancher/rke2-cloud-provider:v1.28.2-build20231016 has already been pulled"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=error msg="Error encountered while importing /var/lib/rancher/rke2/agent/images/cloud-controller-manager-image.txt: failed to pull images from /var/lib/rancher/rke2/agent/images/cloud-controller-manager-image.txt: image \"index.docker.io/rancher/rke2-cloud-provider:v1.28.2-build20231016\": not found"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Pulling images from /var/lib/rancher/rke2/agent/images/etcd-image.txt"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Image index.docker.io/rancher/hardened-etcd:v3.5.9-k3s1-build20230802 has already been pulled"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=error msg="Error encountered while importing /var/lib/rancher/rke2/agent/images/etcd-image.txt: failed to pull images from /var/lib/rancher/rke2/agent/images/etcd-image.txt: image \"index.docker.io/rancher/hardened-etcd:v3.5.9-k3s1-build20230802\": not found"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Pulling images from /var/lib/rancher/rke2/agent/images/kube-apiserver-image.txt"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Image index.docker.io/rancher/hardened-kubernetes:v1.27.12-rke2r1-build20240315 has already been pulled"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=error msg="Error encountered while importing /var/lib/rancher/rke2/agent/images/kube-apiserver-image.txt: failed to pull images from /var/lib/rancher/rke2/agent/images/kube-apiserver-image.txt: image \"index.docker.io/rancher/hardened-kubernetes:v1.27.12-rke2r1-build20240315\": not found"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Pulling images from /var/lib/rancher/rke2/agent/images/kube-controller-manager-image.txt"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Image index.docker.io/rancher/hardened-kubernetes:v1.27.12-rke2r1-build20240315 has already been pulled"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=error msg="Error encountered while importing /var/lib/rancher/rke2/agent/images/kube-controller-manager-image.txt: failed to pull images from /var/lib/rancher/rke2/agent/images/kube-controller-manager-image.txt: image \"index.docker.io/rancher/hardened-kubernetes:v1.27.12-rke2r1-build20240315\": not found"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Pulling images from /var/lib/rancher/rke2/agent/images/kube-scheduler-image.txt"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Image index.docker.io/rancher/hardened-kubernetes:v1.27.12-rke2r1-build20240315 has already been pulled"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=error msg="Error encountered while importing /var/lib/rancher/rke2/agent/images/kube-scheduler-image.txt: failed to pull images from /var/lib/rancher/rke2/agent/images/kube-scheduler-image.txt: image \"index.docker.io/rancher/hardened-kubernetes:v1.27.12-rke2r1-build20240315\": not found"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Pulling images from /var/lib/rancher/rke2/agent/images/runtime-image.txt"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Image index.docker.io/rancher/rke2-runtime:v1.27.12-rke2r1 has already been pulled"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=error msg="Error encountered while importing /var/lib/rancher/rke2/agent/images/runtime-image.txt: failed to pull images from /var/lib/rancher/rke2/agent/images/runtime-image.txt: image \"index.docker.io/rancher/rke2-runtime:v1.27.12-rke2r1\": not found"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Connecting to proxy" url="wss://127.0.0.1:9345/v1-rke2/connect"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Running kubelet --=true --MAX_PODS=250 --address=0.0.0.0 --alsologtostderr=false --anonymous-auth=false --authentication-token-webhook=true --authorization-mode=Webhook --cgroup-driver=systemd --client-ca-file=/var/lib/rancher/rke2/agent/client-ca.crt --cloud-provider=external --cluster-dns=10.43.0.10 --cluster-domain=cluster.local --container-runtime-endpoint=unix:///run/k3s/containerd/containerd.sock --containerd=/run/k3s/containerd/containerd.sock --eviction-hard=imagefs.available<5%,nodefs.available<5% --eviction-minimum-reclaim=imagefs.available=10%,nodefs.available=10% --fail-swap-on=false --feature-gates=CloudDualStackNodeIPs=true --healthz-bind-address=127.0.0.1 --hostname-override=s12142045 --kubeconfig=/var/lib/rancher/rke2/agent/kubelet.kubeconfig --log-file=/var/lib/rancher/rke2/agent/logs/kubelet.log --log-file-max-size=50 --logtostderr=false --node-ip=[SERVER IP REMOVED] --node-labels=cattle.io/os=linux,rke.cattle.io/machine=6decfee3-d1fa-4913-93f1-990639f88862 --pod-infra-container-image=index.docker.io/rancher/mirrored-pause:3.6 --pod-manifest-path=/var/lib/rancher/rke2/agent/pod-manifests --read-only-port=0 --resolv-conf=/run/systemd/resolve/resolv.conf --serialize-image-pulls=false --stderrthreshold=FATAL --tls-cert-file=/var/lib/rancher/rke2/agent/serving-kubelet.crt --tls-private-key-file=/var/lib/rancher/rke2/agent/serving-kubelet.key"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Handling backend connection request [s12142045]"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=error msg="Kubelet exited: exit status 1"
Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Waiting to retrieve kube-proxy configuration; server is not ready: https://127.0.0.1:9345/v1-rke2/readyz: 500 Internal Server Error"
Apr 07 04:52:54 s12142045 rke2[61201]: time="2024-04-07T04:52:54Z" level=error msg="Kubelet exited: exit status 1"
Apr 07 04:52:54 s12142045 rke2[61201]: {"level":"warn","ts":"2024-04-07T04:52:54.288418Z","logger":"etcd-client","caller":"v3@v3.5.9-k3s1/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc0011101c0/127.0.0.1:2379","attempt":0,"error":"rpc error: code = DeadlineExceeded desc = latest balancer error: last connection error: connection error: desc = \"transport: Error while dialing: dial tcp 127.0.0.1:2379: connect: connection refused\""}
Apr 07 04:52:54 s12142045 rke2[61201]: {"level":"info","ts":"2024-04-07T04:52:54.288772Z","logger":"etcd-client","caller":"v3@v3.5.9-k3s1/client.go:210","msg":"Auto sync endpoints failed.","error":"context deadline exceeded"}
Apr 07 04:52:54 s12142045 rke2[61201]: time="2024-04-07T04:52:54Z" level=info msg="Waiting to retrieve kube-proxy configuration; server is not ready: https://127.0.0.1:9345/v1-rke2/readyz: 500 Internal Server Error"
Apr 07 04:52:59 s12142045 rke2[61201]: time="2024-04-07T04:52:59Z" level=error msg="Kubelet exited: exit status 1"
Apr 07 04:52:59 s12142045 rke2[61201]: time="2024-04-07T04:52:59Z" level=info msg="Waiting to retrieve kube-proxy configuration; server is not ready: https://127.0.0.1:9345/v1-rke2/readyz: 500 Internal Server Error"
Apr 07 04:52:59 s12142045 rke2[61201]: time="2024-04-07T04:52:59Z" level=info msg="Pod for etcd not synced (pod sandbox not found), retrying"
```

**Expected Result**

With RKE2 installation script it's working fine.

**Additional context**

**Comment by brandond** [comment]
`Apr 07 04:52:49 s12142045 rke2[61201]: time="2024-04-07T04:52:49Z" level=info msg="Running kubelet --=true --MAX_PODS=250 --address=0.0.0.0 --alsologtostderr=false --anonymous-auth=false --authentication-token-webhook=true --authorization-mode=Webhook --cgroup-driver=systemd --client-ca-file=/var/lib/rancher/rke2/agent/client-ca.crt --cloud-provider=external --cluster-dns=10.43.0.10 --cluster-domain=cluster.local --container-runtime-endpoint=unix:///run/k3s/containerd/containerd.sock --containerd=/run/k3s/containerd/containerd.sock --eviction-hard=imagefs.available<5%,nodefs.available<5% --eviction-minimum-reclaim=imagefs.available=10%,nodefs.available=10% --fail-swap-on=false --feature-gates=CloudDualStackNodeIPs=true --healthz-bind-address=127.0.0.1 --hostname-override=s12142045 --kubeconfig=/var/lib/rancher/rke2/agent/kubelet.kubeconfig --log-file=/var/lib/rancher/rke2/agent/logs/kubelet.log --log-file-max-size=50 --logtostderr=false --node-ip=[SERVER IP REMOVED] --node-labels=cattle.io/os=linux,rke.cattle.io/machine=6decfee3-d1fa-4913-93f1-990639f88862 --pod-infra-container-image=index.docker.io/rancher/mirrored-pause:3.6 --pod-manifest-path=/var/lib/rancher/rke2/agent/pod-manifests --read-only-port=0 --resolv-conf=/run/systemd/resolve/resolv.conf --serialize-image-pulls=false --stderrthreshold=FATAL --tls-cert-file=/var/lib/rancher/rke2/agent/serving-kubelet.crt --tls-private-key-file=/var/lib/rancher/rke2/agent/serving-kubelet.key"
`

> Running kubelet --=true --MAX_PODS=250

This is not a Rancher issue. You have mangled values in your kubelet-args. The kubelet is correctly rejecting these args and exiting with an error message. Please correct the flags you're passing to the kubelet.

**Comment by pamanseau** [comment]
@brandond Thank you for pointing this out.
I'm not passing any flags as you can see in my installation command:
'curl -fL https://xxx.xxx.com/system-agent-install.sh | sudo  sh -s - --server https://xxx.xxx.com --label 'cattle.io/os=linux' --token 929smr7cfjrcmkvvrrnnljzjqn64tgw22ncqt2d9n54hb755mjcz7p --etcd --controlplane --worker`

I think there is an issue in Ranger using the UI to create a RKE2 cluster.
The only parameter that I added in the MAX_PODS=250
![Screenshot 2024-04-11 at 14 52 38](https://github.com/rancher/rancher/assets/1392986/eab9580f-310c-4e1b-9203-d03530006f98)

**Comment by brandond** [comment]
That's not a valid arg. Did you want `--max-pods=250`? 

See the upstream docs for valid kubelet args: https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/.

**Comment by pamanseau** [comment]
@brandond Oh I see my bad. I used to do it like that in Rancher < 2.8.2
Thank you.

**Comment by pamanseau** [comment]
@brandond I will try again but if I look at the logs I see:
```
--=true --MAX_PODS=250
```

This means the -- is already added so I just need to do lowercase.
`max_pods=250`

What is the `--=True` ? This should not be there. Is there a missing validation or something that adds this?
I remember that one time I added an empty argument but I tried and made sure there are no empty ones and it was the same result.
